pkg/crc/config/config.go

```diff
@@ -55,6 +55,11 @@ func (c *Config) Set(key string, value interface{}) (string, error) {
 		return "", fmt.Errorf(configPropDoesntExistMsg, key)
 	}
 
+	ok, expectedValue := c.settingsByName[key].validationFn(value)
+	if !ok {
+		return "", fmt.Errorf(invalidProp, value, key, expectedValue)
+	}
+
 	var castValue interface{}
```

Comment:
Looking at `ValidateBool`, it special cases `true`/`false`, but does not make use of `strconv.ParseBool`. Should both behaviours be unified? It makes more sense to first convert the value to the correct type (casting), and then validate this value. I understand you want a better error message, but I'm not sure exchanging them is the right way of achieving that. Maybe there's a way of 'merging' both steps somehow?

Reply:
@cfergeau But we are casting every user input as part of validation function and if casting fails then returning expected message. in case of `ValidateBool` if a user provide something different than `true/false` string the reason is present which is easy to understand. Now with providing the valuetype of key as part of config help should remove some confusion around it.

Reply:
`ValidateBool` only accepts "true"/"false", contrast this with `cast.ToBoolE` which uses https://golang.org/pkg/strconv/#ParseBool :
>  It accepts 1, t, T, TRUE, true, True, 0, f, F, FALSE, false, False
This is the inconsistency/change in behaviour I wanted to point out.

Reply:
> in case of ValidateBool if a user provide something different than true/false string the reason is present which is easy to understand. Now with providing the valuetype of key as part of config help should remove some confusion around it.

What would be even nicer is to have `memory - memory size in MiB (integer, default '9000')` rather than `memory - number`. I think this could be done with the addition of a help string to `config.Setting`?

Reply:
> Instead of only providing all the details (minimum size, ...) when there's a failure, it would be even nicer is to have memory - memory size in MiB (integer, default '9000') rather than memory - number. I think this could be done with the addition of a help string to config.Setting?

@cfergeau Sure, but that will need bit more work, may be a follow up PR, I can create a tracking issue if you want.

Reply:
@cfergeau created #1933

Reply:
> `ValidateBool` only accepts `true`/`false`, contrast this with `cast.ToBoolE` which uses https://golang.org/pkg/strconv/#ParseBool :
> 
> > It accepts 1, t, T, TRUE, true, True, 0, f, F, FALSE, false, False
> 
> This is the inconsistency/change in behaviour I wanted to point out.

This PR has been merged without further discussion on this point. I'd like to emphasize that there was a command line behaviour change before/after this PR:

Before:
```
$ crc config set skip-check-vsock 1
Successfully configured skip-check-vsock to true
```

After:
```
$ crc config set skip-check-vsock 1
Value '1' for configuration property 'skip-check-vsock' is invalid, reason: must be true or false
```